app/src/main/java/com/example/calculator/MainActivity.kt

```diff
@@ -1,11 +1,144 @@
 package com.example.calculator
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.text.TextUtils
+import android.util.Log
+import android.view.View
+import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import com.example.calculator.databinding.ActivityMainBinding
+
+class MainActivity : AppCompatActivity() , View.OnClickListener{
+
+    private lateinit var binding: ActivityMainBinding
+    private var operandStr = StringBuilder("")
+    private var operand: String = ""
+    private var operandListDefault = arrayListOf("+" , "-" , "x" , "/")
+    private var numberList = arrayListOf("0" , "1" , "2" , "3" , "4" , "5" , "6" , "7" , "8" , "9")
+    private var operandList = ArrayList<String>()
+    private var number1 = 0.0
+    private var number2 = 0.0
+    private var isNewOperand = false
+    private var result = 0.0
 
-class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        this.initView()
+    }
+
+    fun initView() {
+        listOf(binding.bt0 , binding.bt1, binding.bt2 , binding.bt3 , binding.bt4 , binding.bt5 , binding.bt6 , binding.bt7
+        , binding.bt8 , binding.bt9 , binding.btClear , binding.btDot , binding.btEqual , binding.btModulus , binding.btNegative ,
+        binding.btDel , binding.btMulti , binding.btPlus , binding.btSubtract , binding.btDivision)
+                .forEach { it.setOnClickListener(this) }
+    }
+
+    fun showText(btClick: Button) {
+        operandStr.append(btClick.text)
+        binding.tvResult.text = operandStr.toString()
+    }
+
+    override fun onClick(p0: View?) {
+        val btClick = p0 as Button
```

Comment:
cái này check p0 là button trước rồi ép kiểu không lỗi
```suggestion
        val clicked = if (p0 is Button) p0 else return
```